api/src/main/java/net/kyori/adventure/text/serializer/ComponentSerializer.java

```diff
@@ -137,4 +138,39 @@
 
     return this.serialize(component);
   }
+
+  /**
+   * A {@link Component} serializer and deserializer builder.
+   *
+   * @param <I> the input component type
+   * @param <O> the output component type
+   * @param <R> the serialized type
+   *
+   * @since 4.14.0
+   */
+  interface Builder<I extends Component, O extends Component, R> {
```

Comment:
`postProcessor` and `preProcessor` names are confusing here. I guess it's about deserialization? Then it should be named `postDeserialization` and `preDeserialization`. But then it's weird that there's no such methods for serialization.